ambari/tests/test_ambari.py

```diff
@@ -0,0 +1,55 @@
+# (C) Datadog, Inc. 2019
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+from datadog_checks.ambari import AmbariCheck
+
+
+def test_flatten_service_metrics():
+    metrics = AmbariCheck.flatten_service_metrics({"metric_a": 10,
+                                                        "metric_b": 15,
+                                                        "metric_c": {"submetric_c": "hello"},
+                                                        "metric_d": {"submetric_d": {'subsub_d': 25}}
+                                                        }, "pfx")
+    assert metrics == {'pfx.metric_a': 10,
+                        'pfx.metric_b': 15,
+                        'pfx.submetric_c': 'hello',
+                        'pfx.subsub_d': 25
+                        }
+
+
+def test_flatten_host_metrics():
```

Comment:
Hi @agottlie,  this is the old host_metrics_iterate method, I've just renamed it. I have a question, why dont we add the prefix to the nested metrics? Shouldnt it be `pfx.metric_c.submetric_c`?

Reply:
I thought that might be too verbose for metric names, since some of the names are already quite long.  But if you think it makes sense to include the submetrics name (which can go multiple levels sometimes) that should be fine.

Reply:
My question is about the prefix that is missing, this method was already adding the whole path of metrics name

Reply:
In fact another question is why one method is keeping the whole path while the other one isnt